### DISCUSSION TITLE
Unable to use templates option in latest version of td-agent along with opensearch plugin

### DISCUSSION THREAD

**Discussion by vikranth06** [fluent/fluentd#4140] [Q&A]
### Describe the bug

We are using the latest version of td-agent along with Opensearch from AWS but as soon as we add templates section in the config file following error message is thrown "2023-01-16 13:47:39 +0000 [error]: #0 fluent/log.rb:372:error: unexpected error error_class=ArgumentError error="expected both :access_key_id and :secret_access_key options""

### To Reproduce

have the below mentioned config file on the system and the issue will be reproduced.

### Expected behavior

As per the documentation templates is supported. Can the below config file be reviewed and let us know if there is a mistake?

### Your Environment

```markdown
- Fluentd version: 1.15.3
- TD Agent version: 4.4.2
- Operating system: Ubuntu 20.04
- Kernel version:
```

### Your Configuration

```apache
<match **.*******.UPLOAD>
  @type copy
  <store>
    @type opensearch
    index_name xx-xxxx-xxxxx-data
    type_name "_doc"
    logstash_format true
    logstash_prefix xx-xxxx-xxxxx-data
    logstash_dateformat %Y.%m
    template_overwrite true
    templates {"xxxxxxxxxxxxxx_template": "/opt/aws/custom/xxxxx_index_template.json"}
    include_tag_key false
    buffer_type file
    buffer_chunk_limit 2m
    buffer_queue_limit 1000
    buffer_path /var/log/td-agent/uploadreportsupesbuffer
    disable_retry_limit true
    flush_interval 60s
    retry_wait 10s
    id_key MessageId
    reload_connections false
    remove_keys MessageId
    <endpoint>
      url xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
      region us-east-1
      access_key_id xxxxxxxxxxxxxxxxxxxxxxxxxxxxx
      secret_access_key xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
	  </endpoint>
  </store>
</match>
```

### Your Error Log

```shell
"2023-01-16 13:47:39 +0000 [error]: #0 fluent/log.rb:372:error: unexpected error error_class=ArgumentError error="expected both :access_key_id and :secret_access_key options""
```

**Comment by daipom** [comment]
Sorry for the delay.
I moved this issue to discussions as I can't judge this is a bug or not. ([the contributing guidelines](https://github.com/fluent/fluentd/blob/d5df992d844af38233c4d3788de714d644ddb6c4/CONTRIBUTING.md))

**Reply by daipom** [reply]
Note: https://github.com/fluent/fluent-plugin-opensearch/issues/91
